common/lua-system.cpp

```diff
@@ -62,10 +69,36 @@ namespace tec {
 		}
 	}
 
+	void LuaSystem::On(std::shared_ptr<ClientConnectedEvent> data) {
+		CallFunction("onClientConnected");
+	}
+	
+	void LuaSystem::On(std::shared_ptr<ClientDisconnectedEvent> data) {
+		CallFunction("onClientDisconnected");
+	}
+
+	void LuaSystem::CallFunction(std::string function_name) {
+		// multiton <eid, LuaScript*>
+		for (auto itr = LuaScriptMap::Begin(); itr != LuaScriptMap::End(); itr++) {
+			if (!itr->second->script_name.empty() && itr->second->environment[function_name].valid()) {
+				itr->second->environment[function_name]();
+			}
+		}
+		// list<scripts>
+		for (auto script = scripts.begin(); script != scripts.end(); script++) {
+			if (!script->script_name.empty() && script->environment[function_name].valid()) {
+				script->environment[function_name]();
+			}
+		}
+	}
+
 	std::shared_ptr<LuaScript> LuaSystem::LoadFile(FilePath filepath) {
 		std::shared_ptr<LuaScript> script = std::make_shared<LuaScript>(ScriptFile::Create(filepath));
 		script->SetupEnvironment(&this->lua);
 		script->ReloadScript();
+
+		// add to list of scripts
+		scripts.push_back(*script);
```

Comment:
Be cautious here as you are changing the behavior now. I see what your doing, and it is probably the better choice to do what you are doing.

Reply:
It could be stored as a single script instead where by this is avoided. Its comes down to how the API should be setup.